src/sqlfluff/core/linter/linting_result.py

```diff
@@ -181,3 +184,32 @@ def tree(self) -> Optional[BaseSegment]:  # pragma: no cover
                 "path."
             )
         return self.paths[0].tree
+
+    def mark_failed_files_unfixable(self) -> Tuple[int, int]:
+        """Discard lint fixes for files with templating or parse errors.
+
+        Scan all LintedFiles:
+        - Files with none of these errors: No action
+        - Files with these errors (before filtering): Removes fixes from
+          SQLLintError objects, i.e. marks them "unfixable"
```

Comment:
I misread this that we are removing `SQLLintError`s. But actually what we're doing is removing `SQLParseError` and `SQLTemplaterError` from the list of linting errors and leaving the other `SQLLintError`s still. Maybe this could be clearer?

```suggestion
        - Files with these errors (before filtering): Removes fixes for
          `SQLParseError` and `SQLTemplaterError` errors (i.e. marks them
          "unfixable").
```

Reply:
It's not removing fixes for SQLParseError and SQLTemplaterError errors. It's removing fixes for SQLLintError in *files* with SQLParseError and SQLTemplaterError. The SQLParseError and SQLTemplaterError objects are left untouched.

Reply:
I'm changing the `hasattr(violation, "fixes")` to `isinstance(violation, SQLLintError)`. This will make it clearer what's happening. 👍